File: LinearSearch.c
#include <stdio.h>

int main() {
    int haystack[6] = {1, 5, 8, 33, 65, 98};
    int len = sizeof(haystack) / sizeof(haystack[0]);
    int needle = 5;

    for (int i = 0; i < len; i++) {
        if (haystack[i] == needle) {
            printf("Found it!\n");
            return 0;
        }
    }
    printf("Not found\n");
}

File: LinearSearch.py
from typing import List

haystack = [1, 23, 54, 3, 898, 445]
needle = 23


# * Implementing a linear search (Python's "if i in list:" is a linear search)
def linear_search(haystack: List, needle: int) -> List:
    for i in range(len(haystack)):
        if haystack[i] == needle:
            print(f'Found it at {i}')
            return True
        
    print('Not found')
    return False


linear_search(haystack, needle)
        